spring-pulsar/src/main/java/org/springframework/pulsar/core/CachingPulsarProducerFactory.java

```diff
@@ -151,18 +152,50 @@ private void closeProducer(Producer<T> producer) {
 	}
 
 	/**
-	 * Holder for a schema, topic and optional message router used as unique identifier of a producer in cache key.
-	 *
-	 * @param schema schema of the message
-	 * @param topicName topic to send the message to
-	 * @param messageRouter router to use to send the topic
+	 * Uniquely identifies a producer that was handed out by the factory.
 	 *
 	 * @param <T> type of the schema
 	 */
-	record SchemaTopic<T>(Schema<T> schema, String topicName, MessageRouter messageRouter) {
-		public SchemaTopic {
+	static class ProducerCacheKey<T> {
+
+		private final Schema<T> schema;
+		private final SchemaHash schemaHash;
```

Comment:
Unfortunately, having the create a new instance field (`schemaHash`) is not supported in records. Had to go back to hand-written POJO for now.